x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,784 @@
+the next step in this procedure.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `node1-http.p12`.
++
+If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename so that {es} doesn't attempt to reload the file
+before you update the password.
+
+. If you needed to save the new keystore with a new filename, update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore.
+For example:
++
+[source,yaml]
+----
+xpack.security.http.ssl.enabled: true
+xpack.security.http.ssl.keystore.path: node1-http.p12
+----
+
+. If your keystore password is changing, add the password for your private key
+to the secure settings in {es}.
++
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.http.ssl.keystore.secure_password
+----
+
+. Start the node where you updated the keystore, and then complete the remaining
+steps for a <<restart-cluster-rolling,rolling restart>>.
+
+. [[verify-keystore-http-newca,step 10]](Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. One node at a time, complete <<start-rolling-restart-http-newca>> through
+<<verify-keystore-http-newca>> until you've updated all keystores in your cluster.
+
+. (Optional) After rotating keystores on each node in your cluster, remove the
+old CA certificate from your truststore. 
++
+[source,shell]
+----
+keytool -delete -noprompt -alias <old-ca>  -keystore config/http.p12 \
+-storepass <password>
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `-alias`:: Name of the old CA certificate that you want to remove from your
+   truststore.
+====
+
+[discrete]
+[[http-kibana-newca-whatsnext]]
+==== What's next?
+Well done! You've updated the keystore for the HTTP layer. You can now
+<<node-certs-different-kibana,update encryption between {kib} and {es}>>.
+
+[[node-certs-different-kibana]]
+==== Update encryption between {kib} and {es}
+
+When you ran the `elasticsearch-certutil` tool with the `http` option, it
+created a `/kibana` directory containing an `elasticsearch-ca.pem` file. You
+use this file to configure {kib} to trust the {es} CA for the HTTP
+layer.
+
+. Copy the `elasticsearch-ca.pem` file to the {kib} configuration directory,
+as defined by the `KBN_PATH_CONF` path.
++
+NOTE: `KBN_PATH_CONF` contains the path for the {kib} configuration files. If
+you installed {kib} using archive distributions (`zip` or `tar.gz`), the
+path defaults to `KBN_HOME/config`. If you used package distributions
+(Debian or RPM), the path defaults to `/etc/kibana`.
+
+. If you modified the filename for the `elasticsearch-ca.pem` file, edit
+`kibana.yml` and update the configuration to specify the location of the
+security certificate for the HTTP layer.
++
+[source,yaml]
+----
+elasticsearch.ssl.certificateAuthorities: KBN_PATH_CONF/elasticsearch-ca.pem
```

Comment:
for instance here we can put the new CA file along with the old one first ,then make the changes in ES in all nodes, and then remove the old CA certificate. `elasticsearch.ssl.certificateAuthorities` supports an array of values